lib/core/macros.nim

```diff
@@ -1611,3 +1611,43 @@ when defined(nimMacrosSizealignof):
     ## from a field of a type. Therefore it only requires one argument
     ## instead of two. Returns a negative value if the Nim compiler
     ## does not know the offset.
+
+proc splitDefinition*(def: NimNode): tuple[lhs: NimNode, rhs: NimNode] =
+  ## allows library constructs such as: `byRef: a2 = expr`
+  doAssert def.kind == nnkStmtList and def.len == 1
+  let def2 = def[0]
+  doAssert def2.kind == nnkAsgn
+  let lhs = def2[0]
+  let rhs = def2[1]
+  expectKind(lhs, nnkIdent)
+  return (lhs, rhs)
+
+macro byRef*(def: untyped): untyped =
+  ## Defines a ref alias for lvalue expressions. The expression is evaluated
+  ## only once, and any side effects will only be evaluated once, at declaration
+  ## time.
+  runnableExamples:
```

Comment:
See? You are misusing ``runnableExamples``, this is a test, not an example!

Reply:
Yes, please stop doing this. Tests need to go at the bottom of a file under `when isMainModule` or in a test file in the ``tests/`` dir.

Reply:
fixed: moved tests to tests/stdlib/tsugar.nim and the new `runnableExamples` is now as minimal as possible (both asserts are helpful in `doAssert type(x1) is int and x == @[1,12,3]` for doc purposes)